# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../pkgconfig/pkg-config-0.23-sysroot.patch
# Copyright (C) 2008 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

http://lists.freedesktop.org/archives/pkg-config/2008-January/000255.html
pkg-config 0.23 PKG_CONFIG_SYSROOT_DIR bug
Paul Bender <pebender@san.rr.com>
Tue Jan 22 20:04:41 PST 2008

There appears to be a bug in the handling of the new
PKG_CONFIG_SYSROOT_DIR. When PKG_CONFIG_SYSROOT_DIR is set, on the all
but the -I and -L options are removed from --cflags and --libs
respectively. This does not seem like the right behavior.

I have attached a patch that fixes the problem for me.

diff -Naur pkg-config-0.23-old/pkg.c pkg-config-0.23-new/pkg.c
--- pkg-config-0.23-old/pkg.c	2008-01-16 13:59:49.000000000 -0800
+++ pkg-config-0.23-new/pkg.c	2008-01-22 15:00:55.000000000 -0800
@@ -483,6 +483,10 @@
 	      g_string_append (str, pcsysrootdir);
 	      g_string_append (str, tmpstr+2);
 	    }
+          else 
+	    {
+	      g_string_append (str, tmpstr);
+	    }
 	}
       else 
 	{
